src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+                Actor_SetDropFlag(&this->actor, &this->collider.info);
+                Enemy_StartFinishingBlow(play, &this->actor);
+                this->actor.flags &= ~ACTOR_FLAG_1;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F8E0.s")
+            if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_BLUNT) {
+                EnSlime_ReactToBluntHit(this);
+            } else if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_STUN) {
+                this->timer = 40;
+                Actor_SetColorFilter(&this->actor, 0, 255, 0x2000, 40);
+                Actor_PlaySfx(&this->actor, NA_SE_EN_COMMON_FREEZE);
+                EnSlime_EnterStunnedState(this);
+            } else if (this->actor.colChkInfo.damageEffect != ENSLIME_DMGTYPE_HOOKSHOT) {
+                if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_ICEARROW) {
+                    if (this->actor.params == EN_SLIME_TYPE_BLUE) {
+                        EnSlime_InitIceBlock(this);
+                    } else {
+                        EnSlime_InitIceStun(this);
+                        if (this->actor.colChkInfo.health == 0) {
+                            this->timer = 3;
+                            this->collider.base.acFlags &= ~AC_ON;
+                        }
+                        EnSlime_EnterStunnedState(this);
+                    }
+                } else {
+                    if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_LIGHT) {
+                        this->effectAlpha = 4.0f;
+                        this->effectScale = 0.4f;
+                        this->damageEffectType = ACTOR_DRAW_DMGEFF_LIGHT_ORBS;
+                        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->collider.info.bumper.hitPos.x,
+                                    this->collider.info.bumper.hitPos.y, this->collider.info.bumper.hitPos.z, 0, 0, 0,
+                                    4);
+                    } else if (this->actor.colChkInfo.damageEffect == ENSLIME_DMGTYPE_ELECTRIC) {
+                        this->damageEffectType = ACTOR_DRAW_DMGEFF_ELECTRIC_SPARKS_LARGE;
+                        this->effectAlpha = 4.0f;
+                        this->effectScale = 0.4f;
+                    }
+                    EnSlime_ReactToDamage(this, play, true);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F9A0.s")
+void EnSlime_Update(Actor* thisx, PlayState* play) {
+    EnSlime* this = THIS;
+    s32 pad;
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FA88.s")
+    EnSlime_CheckACHit(this, play);
+    this->actionFunc(this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FB60.s")
+    // Note: Does not match if you use this->actor instead of thisx (iirc this is true for the whole function for some
+    // reason)
```

Comment:
I would just cut out the parenthetical statement; this is common enough that all you really need to say is it doesn't match if you use `this`.